debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400DatabaseSchema.java

```diff
@@ -56,8 +57,8 @@ public void addSchema(Table table) {
         TableId id = table.id();
         // store in the short name
         // used directly by the decoder
-        final String systemTableName = jdbcConnection.getSystemName(id.schema(), id.table());
-        map.put(id.catalog() + id.schema() + systemTableName, tableInfo);
+        final Optional<String> systemTableNameOpt = jdbcConnection.getSystemName(id.schema(), id.table());
+        systemTableNameOpt.map(systemTableName -> map.put(id.catalog() + id.schema() + systemTableName, tableInfo));
```

Comment:
This seems a little strange to me. I could understand how the table name might not be present, so we can't convert it to the system name.

But then, wouldn't we expect something earlier (`readTableStructure`?) to have failed? I'm not sure how we'd be able get the format of a table without also being able to retrieve its system name.

Reply:
you are right it wont find the structure but this is using the same method call that now returns an option for the lookup so for the code to compile it also needs to change. I could probably be more aggressive and move the loopup up and put everything else in the map